targets/TARGET_STM/TARGET_STM32F7/flash_api.c

```diff
@@ -167,6 +174,9 @@ int32_t flash_program_page(flash_t *obj, uint32_t address, const uint8_t *data,
         }
     }
 
+    SCB_CleanInvalidateDCache_by_Addr(&address, size);
```

Comment:
Again, wants `(uint32_t *) address`, not `&address`.

Except the loop above has incremented/decremented them, so you'll need to do something to store the original values.